### DISCUSSION TITLE
slidev-theme-neversink

### DISCUSSION THREAD

**Discussion by gureckis** [slidevjs/slidev#1764] [Show and tell]
I created a Slidev theme called [Neversink](https://gureckis.github.io/slidev-theme-neversink/) well suited for academic/education presentations for computational/mathematical topics.  Why might it interest you?

- It has a couple custom layouts which help you not have to do as much CSS stuff for layouts including a [simple design system for aligning two column materials](https://gureckis.github.io/slidev-theme-neversink/layouts/two-cols-title.html).  For example to make a two column slide with a title:

```
---
layout: two-cols-header
columns: is-4
align: c-rm-lt
color: light
titlepos: t
---

:: title ::

# My title

:: left ::

This is the left column

:: right ::

This is the right column


```

This will devote 4/12 units to the left column, align the title, right-middle align the left column, left-top align the right column, change the color scheme to `light` and put the title at the top.  Most of these things can be done in CSS but it's nice to have a simple way to specify these things in frontmatter.

- It has some simple [monochrome color schemes](https://gureckis.github.io/slidev-theme-neversink/colors.html) that apply at the slide and element level.

- Include some fun elements like a [movie scrolling credits](https://gureckis.github.io/slidev-theme-neversink/layouts/credits.html), [speech bubbles](https://gureckis.github.io/slidev-theme-neversink/components/speechbubble.html), [sticky notes](https://gureckis.github.io/slidev-theme-neversink/components/stickynote.html), and [admonition](https://gureckis.github.io/slidev-theme-neversink/components/admonitions.html).

Happy for suggestions and PRs!

**Comment by kermanx** [comment]
Wow, it's such a nice theme! Could you help make a PR to add your theme to https://github.com/slidevjs/slidev/blob/main/docs/.vitepress/themes.ts#L122? Then more people will see it.

**Reply by zillionare** [reply]
I have also make a theme, which includes many features such as admonition, cards, markmap, device mockup, timeline and more, and it's blocked from merging. I wonder why it's not accepted. It's called rich because it can provide very rich UI components

https://github.com/slidevjs/slidev/pull/1728

https://zillionare.github.io/slidev-theme-rich

**Reply by zillionare** [reply]
Really love this one. Maybe you can refer to https://zillionare.github.io/slidev-theme-rich to add these features?

1. timeline it's a very common story telling feature in presentation
<img width="462" alt="image" src="https://github.com/user-attachments/assets/b9cd2467-898d-4f42-ade1-c9d24838fd62">

3. markmap maybe
4. more text effects like animated.css or SoarText
5. Device mock like this:
<img width="851" alt="image" src="https://github.com/user-attachments/assets/af65706f-f990-40cb-b075-b239c145ea93">

I myself doesn't good at web development, so it takes me much more time to develop it, and less robust/beautiful then yours.

By the way, I like the idea of color schema. I'm trying follow these articles to make a new theme called slidev-theme-one, which allows user change color themes:

https://uxdesign.cc/how-to-define-color-usage-through-semantic-sets-for-design-systems-99445804233d

https://medium.com/getaround-eu/colors-that-make-sense-505d0f3509c1

https://buninux.com/learn/how-to-create-a-better-ui-color%C2%A0palette

Not sure if you have already done the same.

**Comment by zillionare** [comment]
by the way, I like StickNotes in this theme. Wonder why we have to enable specific theme to use a component like StickNote. It should be available for every theme.

Make these components either builtin, or built a marketplace to let others contribute. Let theme be just color-schema, then layout controls grids, to be composite of blocks, and components as build blocks. In this way, I think tools like powerpoint can be fully replaced.

**Reply by kermanx** [reply]
We have addons for this: 
https://sli.dev/guide/theme-addon#use-addon and https://sli.dev/guide/write-addon

**Reply by zillionare** [reply]
thanks @KermanX I do find a useful & interesting addon: slidev-addon-rabbit
It's just that there are too few addons

**Reply by gureckis** [reply]
Agree the stickies and admonitions should be add-ons.  However, i wasn't sure right away what features would be worth that effort so put lots of things into the theme right now and will see what i/others uses and possibly refactor those into add ons later.

**Comment by ninadpchaudhari** [comment]
Thank you for sharing :)
I recently transitioned from Marp to Slidev, and your docs really helped me out while creating slides for several of my lab sessions. [link to one of my lab websites if curious!](https://ninad.me/UnIJ0)

One thing I’ve wondered is why the [official Slidev docs on layouts](https://sli.dev/builtin/layouts) don’t include diagrams for the built-in layouts....
Something like what your docs have or even on Bootstrap-style visual guide, would really help new devs. 

Hoping I can carve out some time this summer to contribute to that and towards separating those  stickies and admonitions into addons!
